src/pe.cc

```diff
@@ -0,0 +1,320 @@
+// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bloaty.h"
+
+ABSL_ATTRIBUTE_NORETURN
+static void Throw(const char* str, int line) {
+  throw bloaty::Error(str, __FILE__, line);
+}
+
+#define THROW(msg) Throw(msg, __LINE__)
+#define THROWF(...) Throw(absl::Substitute(__VA_ARGS__).c_str(), __LINE__)
+#define WARN(x) fprintf(stderr, "bloaty: %s\n", x);
+
+using absl::string_view;
+
+namespace bloaty {
+namespace pe {
+const uint16_t dos_magic = 'ZM';
+
+//! Sizes in bytes of various things in the COFF format.
+namespace STRUCT_SIZES {
+enum {
+  Header16Size = 20,
```

Comment:
Why define these at all if we can just use `sizeof(pe_section)`, etc?

Reply:
For the static asserts below.

Reply:
What is the purpose of asserting that these structs are a certain size? Are you trying to catch bugs in `third_party/lief_pe/pe_structures.h`? I think we can presume that the structures defined there are correct without double-checking them here.

Reply:
Since we took a header file out of context, I'd like to verify that struct alignment is still correct.
We are using a different build system, so compiler flags / options might influence this.